# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/midi/MIDIPortChild.h
# Commit: 225adb498b09
# Full Hash: 225adb498b091a93ddf7081c1f4356f705594a7b
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-04-12 15:57:11
# Regressor Bug: 1758468
# File Overlap Count: 1
# Description:
#   Bug 1758468 - Use stable, anonymous IDs for Web MIDI ports r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142550
# ==============================================================================

diff -r 3c6e6d0d7b38 -r 225adb498b09 dom/midi/MIDIPortChild.h
--- a/dom/midi/MIDIPortChild.h	Tue Apr 12 12:31:27 2022 +0000
+++ b/dom/midi/MIDIPortChild.h	Tue Apr 12 12:31:27 2022 +0000
@@ -7,8 +7,8 @@
 #ifndef mozilla_dom_MIDIPortChild_h
 #define mozilla_dom_MIDIPortChild_h
 
+#include "mozilla/dom/MIDIPortInterface.h"
 #include "mozilla/dom/PMIDIPortChild.h"
-#include "mozilla/dom/MIDIPortInterface.h"
 
 namespace mozilla::dom {
 
@@ -33,6 +33,8 @@
 
   MIDIPortChild(const MIDIPortInfo& aPortInfo, bool aSysexEnabled,
                 MIDIPort* aPort);
+  nsresult GenerateStableId(const nsACString& aOrigin);
+  const nsString& StableId() { return mStableId; };
   // virtual void Shutdown() override;
   void SetActorAlive();
 
@@ -43,6 +45,7 @@
   // Pointer to the DOM object this actor represents. The actor cannot outlive
   // the DOM object.
   MIDIPort* mDOMPort;
+  nsString mStableId;
   bool mActorWasAlive;
 };
 }  // namespace mozilla::dom